### PR TITLE
Example for OTLP gRPC exporter for Metrics.

### LINKED ISSUE #1593: OTLP gRPC metrics exporter fails to connect
Hi,

I was trying to use the metrics gRPC exporter, but I can't seem to get it to work.

I'm getting a connection error:
```
[Error] File: [OMITTED]/opentelemetry-cpp-src/exporters/otlp/src/otlp_grpc_metric_exporter.cc:151[OTLP METRIC GRPC Exporter] Export() failed: failed to connect to all addresses
```
The collector service is running, and I'm able to export traces with the same setup. Similarly, I'm able to export metrics using the OTLP HTTP exporter.

Is there anything else that's needed to get the gRPC exporter working, or this is bug?

### PR DESCRIPTION
**Pull request by lalitb**
Fixes #1593 

## Changes
- Adds OTLP GRPC exporter example
- Updates example README.md for metrics, and to use latest version of Collector
- Update Collector config to by in sync with latest Collector version.
- 
Please provide a brief description of the changes here.

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [x] Unit tests have been added
* [x] Changes in public API reviewed

### CODE REVIEW

**Review by esigo — APPROVED** [review]
LGTM
Thanks for the PR

File: examples/otlp/CMakeLists.txt
Comment (owent) [inline examples/otlp/CMakeLists.txt:47]: I think we couldn't link gRPC::grpc++ directly, ther may be some problem as #1603

Comment (lalitb) [reply examples/otlp/CMakeLists.txt:47]: Thanks. Will wait for #1606 to be merged, and incorporate the changes accordingly.

File: examples/otlp/grpc_metric_main.cc
Comment (owent) [inline examples/otlp/grpc_metric_main.cc:36]: Could we use `OtlpGrpcMetricExporterFactory` after #1606  be merged? Just like other examples?
It think we also need factories for metrics SDKs, though this is not a problem of this PR.

File: examples/otlp/CMakeLists.txt
Comment (lalitb) [reply examples/otlp/CMakeLists.txt]: Have removed gRPC::grpc++ now. Thanks.

File: examples/otlp/grpc_metric_main.cc
Comment (lalitb) [reply examples/otlp/grpc_metric_main.cc]: Yes, using OtlpGrpcMetricExporterFactory now.  For factories for metrics SDKs, we have #1485. This is supposed to be done once SDK is nearing GA.

**Review by owent — APPROVED** [review]
LGTM

**Pull request merged**
